WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShippingAddresses/WooShippingEditAddressView.swift

```diff
@@ -0,0 +1,261 @@
+
+    /// Tracks the focused address field.
+    @FocusState private var focusedField: AddressField?
+
+    @Environment(\.dismiss) private var dismiss
+
+    var body: some View {
+        ScrollView {
+            VStack(spacing: Constants.verticalSpacing) {
+                AddressTextField(field: .name, text: $name, focused: $focusedField)
+                if showCompanyField {
+                    AddressTextField(field: .company, text: $company, focused: $focusedField)
+                } else {
+                    Button {
+                        withAnimation {
+                            showCompanyField = true
+                        }
+                    } label: {
+                        Text(Localization.addCompany)
+                    }
+                    .buttonStyle(PlusButtonStyle())
+                    .font(.subheadline)
+                    .bold()
+                }
+                AddressSelection(field: .country, selected: country) {
+                    // TODO: Handle country selection
+                }
+                .padding(.top, Constants.extraPadding)
+                AddressTextField(field: .address, text: $address, focused: $focusedField)
+                AddressTextField(field: .city, text: $city, focused: $focusedField)
+                AdaptiveStack(horizontalAlignment: .leading, verticalAlignment: .top, spacing: Constants.innerSpacing) {
+                    AddressSelection(field: .state, selected: state) {
+                        // TODO: Handle state selection
+                    }
+                    AddressTextField(field: .postalCode, text: $postalCode, focused: $focusedField)
+                }
+                .padding(.bottom, Constants.extraPadding)
+                AddressTextField(field: .email, text: $email, focused: $focusedField)
+                AddressTextField(field: .phone, text: $phone, focused: $focusedField)
+                    .padding(.bottom, Constants.extraPadding)
+                Toggle(Localization.defaultAddress, isOn: $saveAsDefault)
+                    .font(.subheadline)
+                    .tint(Color(.accent))
+            }
+            .padding()
+            .toolbar {
+                ToolbarItem(placement: .cancellationAction) {
+                    Button(Localization.cancel) {
+                        dismiss()
+                    }
+                }
+            }
+        }
+    }
+
+    private struct AddressTextField: View {
+        /// Which address field to display.
+        let field: AddressField
+
+        /// The text to display in the text field.
+        @Binding var text: String
+
+        /// The focused state of the field.
+        @FocusState.Binding var focused: AddressField?
+
+        var body: some View {
+            VStack(spacing: Constants.innerSpacing) {
+                HStack {
+                    Text(field.title)
+                    if field.required {
+                        Text("*")
+                    }
```

Comment:
Should we maybe have explicit `spacing` for the HStack when we have "*" as required mark, or to add " *" to the text string to have the spacing same as in Figma?
On Figma it appears that the "*" is part of the string itself so it has a bit less spacing.

<img width="554" alt="Screenshot 2025-01-09 at 22 04 25" src="https://github.com/user-attachments/assets/ffa0acc8-5053-4736-8ee7-4c16051e6832" />

Reply:
Sure thing, added explicit spacing in 4726b32. I didn't put this in the text string because conceptually the label and whether the field is required are separate. This way when field requirements change (e.g. when the state isn't required for the selected country) we don't have to handle two separate localized strings.

Design|Actual
-|-
![image](https://github.com/user-attachments/assets/81bf7056-295e-4fda-aeac-0a2adb41e7fd)|![Simulator Screenshot - iPhone 16 Pro - 2025-01-10 at 11 08 21](https://github.com/user-attachments/assets/33c7af7c-5362-4c81-9358-67b2e362f4ec)